lambda/getSchedule/tias-app/Scheduler/src/main/java/db/Person.java

```diff
@@ -0,0 +1,208 @@
+            boolean administrator, boolean professor) {
+        this.personId = personId;
+        this.email = email;
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.desiredNumberAssignments = desiredNumberAssignments;
+        this.peerTeacher = peerTeacher;
+        this.teachingAssistant = teachingAssistant;
+        this.administrator = administrator;
+        this.professor = professor;
+        this.availabilities = new ArrayList<Availability>();
+        this.preferences = new ArrayList<Preference>();
+        this.qualifications = new ArrayList<Qualification>();
+
+        availableSections = new ArrayList<>();
+        availabilityScore = 0.f;
+        currentAssignments = 0;
+    }
+
+    public int getPersonId() {
+        return personId;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public int getDesiredNumberAssignments() {
+        return desiredNumberAssignments;
+    }
+
+    public boolean isPeerTeacher() {
+        return peerTeacher;
+    }
+
+    public boolean isTeachingAssistant() {
+        return teachingAssistant;
+    }
+
+    public boolean isAdministrator() {
+        return administrator;
+    }
+
+    public boolean isProfessor() {
+        return professor;
+    }
+
+    public ArrayList<Availability> getAvailabilities() {
+        return (ArrayList<Availability>) Collections.unmodifiableList(availabilities);
+    }
+
+    public void addAvailability(Availability availability) {
+        availabilities.add(availability);
+    }
+
+    public ArrayList<Preference> getPreferences() {
+        return preferences;
+    }
+
+    public void addPreference(Preference preference) {
+        preferences.add(preference);
+    }
+
+    public void sortPreferences() {
+        Collections.sort(preferences, Collections.reverseOrder());
+    }
+
+    public ArrayList<Qualification> getQualifications() {
+        return (ArrayList<Qualification>) Collections.unmodifiableList(qualifications);
+    }
+
+    public void addQualification(Qualification qualification) {
+        qualifications.add(qualification);
+    }
+
+    public float getAvailabilityScore() {
+        return availabilityScore;
+    }
+
+    public ArrayList<Integer> getAvailabileSections() {
+        return (ArrayList<Integer>) Collections.unmodifiableList(availableSections);
+    }
+
+    public int getCurrentAssignments() {
+        return currentAssignments;
+    }
+    
+    public void addCurrentAssignment() {
+        currentAssignments++;
+    }
+
+    boolean isQualified(int courseId) {
+        for (Qualification qualification : qualifications) {
+            if (courseId == qualification.getCourseId()) {
+                return qualification.isQualified();
+            }
+        }
+        return false;
+    }
```

Comment:
**TODO**: Improve from *O(n)* to *O(1)* by converting `qualifications` from `ArrayList` to `HashSet`